bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java

```diff
@@ -709,21 +711,21 @@ private void postAlarms(@Nullable ArrayList<@Nullable String> errors) throws She
         }
     }
 
-    protected @Nullable ArrayList<ShellySettingsInput> fillInputSettings(ShellyDeviceProfile profile,
+    protected @Nullable ArrayList<@Nullable ShellySettingsInput> fillInputSettings(ShellyDeviceProfile profile,
             Shelly2GetConfigResult dc) {
         if (dc.input0 == null) {
             return null; // device has no input
         }
 
-        ArrayList<ShellySettingsInput> inputs = new ArrayList<>();
+        ArrayList<@Nullable ShellySettingsInput> inputs = new ArrayList<>();
```

Comment:
```suggestion
        List<@Nullable ShellySettingsInput> inputs = new ArrayList<>();
```

Reply:
Not possible because the caller expects an `ArrayList` for setting this in the DTO class.